resolwe_bio/processes/differential_expression/edgeR.yml

```diff
@@ -82,8 +82,9 @@
 
       {% for c in control %}
         {% for t in case %}
-          {% if control.0.source != c.source or c.source != t.source %}
-            re-error "DE analysis must be crated using the expression data from the same source."
+          {% if (control|first).source != c.source or c.source != t.source %}
+            re-warning "Genes in all expression data must be annotated by the same genome database."
+            re-error "Sample {{ c|name }} has {{ c.source }} gene IDs, while {{ t|name }} has {{ t.source }}."
```

Comment:
Missing error message for the first part of the condition in the `if` statement.